### PR TITLE
Adding settings for mix_rate, stereo and format for Mic Record Demo 4.0-dev

### LINKED ISSUE #470: Setting of format, mix_rate and stereo flag on mic_record
**Which demo project is affected:**
godot-demo-projects/audio/mic_record

**Description:**
The demo and [instructions](https://docs.godotengine.org/en/stable/tutorials/audio/recording_with_microphone.html?highlight=record) show how to print out the format, mix_rate and stereo flags, but not how to set any of them.  

The [AudioStreamSample](https://docs.godotengine.org/en/stable/classes/class_audiostreamsample.html#class-audiostreamsample) class has a format option FORMAT_IMA_ADPCM, which would compress the signal.  Even being able to set the stereo flag to false would save 50% of the bandwidth, as it's generally not relevant when you have a single microphone and therefore should be part of the example.

(If there is no way to set these values, maybe this issue should be filed with the core system.)

### PR DESCRIPTION
**Pull request by voylin**
Closes #470

### CODE REVIEW

**Comment by voylin** [comment]
I've been trying so many things now for this Pull Request and the checks keep failing. The message said do a rebase or amend but upon doing that, nothing changes ... Quite new to git and github so I'm probably doing something wrong here. But I can't tell what

**Comment by voylin** [comment]
If anyone knows what's going wrong here, please tell me as I have no idea and it's getting quite annoying

**Comment by aaronfranke** [comment]
@voylin See these resources for how to use Git when contributing to Godot:

https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html

https://www.youtube.com/watch?v=0XKy8YpP5Ww

You can also fix CI checks by running the formatting script: `file_format.sh`. If you're on Windows you need WSL to run it.

**Comment by voylin** [comment]
@aaronfranke Thanks for the resources, I figured out that it was just because of some enters that it could not work. The video is especially helpful. Because of you I started using Gitkraken and I could also finally solve the "commit mess" and squash/rebase everything correctly. Thank you very much and sorry for the trouble I've been causing.

File: audio/mic_record/README.md [hunk 1]
```diff
@@ -3,9 +3,12 @@
 This is an example showing how one can record audio from
 the microphone and later play it back or save it to a file.
 
+With an addition on how to change the format, mix rate and
+stereo settings of the recording.
+
 Language: GDScript
 
-Renderer: GLES 2
+Renderer: Vulkan
```
Comment (aaronfranke) [inline audio/mic_record/README.md:11]: If we're updating this text, we should be specifying if it's Vulkan Clustered or Vulkan Mobile.

Additionally, we should be setting demos to Vulkan Mobile if it doesn't look any different compared to Vulkan Clustered. Especially for GUI-only demos, they will look the same in Vulkan Mobile. Go to Project -> Project Settings -> General tab -> Rendering section -> Vulkan -> Back End -> Select Forward Mobile.

**Review by aaronfranke — COMMENTED** [review]
Most of this looks good.

File: audio/mic_record/project.godot [hunk 2]
```diff
@@ -24,13 +24,7 @@ enable_audio_input=true
 
 [display]
 
-window/stretch/mode="2d"
+window/stretch/mode="canvas_items"
 window/stretch/aspect="expand"
 window/size/width=640
 window/size/height=480
```
Comment (aaronfranke) [inline audio/mic_record/project.godot:30]:
```suggestion
window/size/viewport_width=640
window/size/viewport_height=480
```

Comment (voylin) [reply audio/mic_record/project.godot]: Thanks! Sorry about making this mistake multiple times. Fixed, squashed and pushed!

**Review by aaronfranke — APPROVED** [review]
Thanks!

**Pull request merged**